### PR TITLE
Patches to use replication_port, and add replication_options

### PR DESCRIPTION
**Pull request by jayjanssen**
replication_port was not being used in the CHANGE MASTER, that is fixed.

I also added a 'replication_options' option so I can provide arbitrary arguments to the CHANGE MASTER, for example enabling SSL replication:

replication_options=", MASTER_SSL=1, MASTER_SSL_CA='/etc/mysql/CA.pem', MASTER_SSL_CERT='/etc/mysql/server.pem', MASTER_SSL_KEY='/etc/mysql/server.key'"

The only messy bit is the preceding comma.

### CODE REVIEW

File: heartbeat/mysql [hunk 1]
```diff
@@ -582,6 +593,10 @@ check_slave() {
             set_reader_attr 0
             $CRM_MASTER -v 0
 
+            # try to restart slave
```
Comment (y-trudeau) [inline heartbeat/mysql:596]: Have you added these lines?  I ask because I think they were removed recently.

Comment (jayjanssen) [inline heartbeat/mysql:596]: Yves, those lines are in your master in this repo: 

https://github.com/y-trudeau/resource-agents-prm/blob/master/heartbeat/mysql#L58

Comment (jburnham) [inline heartbeat/mysql:596]: It seems when you merged in the changes from https://github.com/jayjanssen/resource-agents-prm/commit/913ad8a9ef2d714ba7a6900a4cab0489d0adcd61 it didn't merge correctly. These lines were removed and put in the if else block right below it to see if we can flock the lockfile. See https://github.com/y-trudeau/resource-agents-prm/commit/beb464fad781d7cda0b28f7870b4a26219f6e44e line 561

Comment (jayjanssen) [inline heartbeat/mysql:596]: Bad link, I see those lines right here: https://github.com/y-trudeau/resource-agents-prm/blob/master/heartbeat/mysql#L582

Comment (jburnham) [inline heartbeat/mysql:596]: The lines to start the slave were moved into the "if type flock" block to support xtrabackup for issue jayjanssen/Percona-Pacemaker-Resource-Agents#3 

See https://github.com/y-trudeau/resource-agents-prm/blob/master/heartbeat/mysql#L586-L600

The start slave that is in your pull request looks like it was from a merge conflict that was resolved incorrectly. The start slave line should not be there and instead is already in the block I mentioned earlier.

**Pull request merged**